src/automaton.rs

```diff
@@ -41,10 +43,49 @@ pub trait Automaton {
     ///
     /// If this returns `true` even when no match is possible, then behavior
     /// will be correct, but callers may be forced to do additional work.
-    fn can_match(&self, state: &Self::State) -> bool;
+    fn can_match(&self, _state: &Self::State) -> bool {
+        true
+    }
+
+    /// Returns true if and only if `state` matches and must match no matter what
+    /// steps are taken.
+    ///
+    /// If this returns `true`, then every sequence of inputs from this state
+    /// produces a match. If this does not follow, then those match states may
+    /// never be reached. In other words, behavior may be incorrect.
+    ///
+    /// If this returns `false` even when every sequence of inputs will lead to
+    /// a match, then behavior will be correct, but callers may be forced to do
+    /// additional work.
+    fn must_match(&self, _state: &Self::State) -> bool {
```

Comment:
Hmm, I feel like a better name for this method would be `will_always_match` or something. `must_match` is crossing some wires in my brain.